frontend/apps/reader/modules/readermenu.lua

```diff
@@ -262,7 +262,7 @@ function ReaderMenu:exitOrRestart(callback)
     end
 end
 
-function ReaderMenu:onShowReaderMenu()
+function ReaderMenu:onShowReaderMenu(tab_index)
```

Comment:
That's fine.
But you should allow for `tab_index` to be `nil` as it was allowed before, and in that case, use `self.last_tab_index` as before. 
So, just add here:
```lua
if not tab_index then
    tab_index = self.last_tab_index
end
```
Because it will be called without a tab_index on device with a Menu key:
https://github.com/koreader/koreader/blob/e3895102578ddb932d3eb44ff155a97cb325a5ec/frontend/apps/reader/modules/readermenu.lua#L60-L62
(It seemed to me between some or your last commits you somehow fixed it in the else of if Device:isTouchDevice(), but I don't see it anymore. Anyway, the above suggestion is probably the proper way to fix it and stay compatible with the other part of the code that may send some Event:new("ShowReaderMenu") without any argument (even if it seems there are no others :)

Reply:
My bad!, don't know how I didn't see that, thank you :), I will stick with the nil checking, as it's proper in my opinion too.